app/main/presenters/search_presenters.py

```diff
@@ -182,21 +188,24 @@ def build_lots_and_categories_link_tree(framework, lots, category_filter_group,
     Returns a list of selected filters: there will always be at least one element, which is
     the root of the tree; there may then be a category; there may then follow a sub-category.
 
-    :param framework: the latest G-Cloud framework from the API
+    :param framework: the framework from the API
     :param lots: a sequence of lot dicts applicable to the live framework(s)
     :param category_filter_group: a single filter group loaded from the framework search_filters manifest
-    :param request: current request so that we can figure out what lots and categories are selected
-    :param content_manifest: a ContentManifest instance for G-Cloud search_filters.
-    :return: list of selected category and lot filters, starting with the 'all categories' root node node
+    :param request: the original request for creating links for nodes
+    :param cleaned_request_args: current cleaned request args so that we can figure out selected lots and categories
+    :param content_manifest: a ContentManifest instance for the frameworks search_filters.
+    :param search_link_builder: a werkzeug Href object for the view category tree links link to
+    :return: list of selected category and lot filters, starting with the 'all categories' root node
     """
-    current_lot_slug = get_lot_from_args(request.args, [lot['slug'] for lot in lots])
+    current_lot_slug = get_lot_from_args(cleaned_request_args, [lot['slug'] for lot in lots])
```

Comment:
This method name (`get_lot_from_args`) is not very descriptive.
What we're actually doing is validating the supplied lot name agains the supplied lots.

At first glance I'm like 'Why do we need a method to `get_lot_from_args` when we can just do `cleaned_request_args.get('lot')` ?'
Then digging in we see that aside from getting the lot from the args it validates the lot against the list that is passed in and defaults to `None`.

Reply:
May be out of scope for this PR but it's really difficult to read what's going on in this `build_lots_and_categories_link_tree ` method

Reply:
So this is the currently selected lot slug. This would also benefit from a more descriptive name. As someone who doesn't know this code well `current` adds nothing to my understanding. It may as well just be `lot_slug`. I can't tell if it's parsed from a url or what kind of operation wee might be performing. Whereas `currently_selected_lot_slug` at least indicates that it is the lot slug that has been selected for an action. Maybe something like `root_lot_slug` or `selected_root_lot_slug` I don't know. just on't think `current_lot_slug` is descriptive enough.

Reply:
Have renamed `get_lot_from_args` to `get_valid_lot_from_args_or_none`

Reply:
Remember that a lot of this stuff is just inherited from the catalogue search, so any improvements made here should be reflected in that code too so the consolidation stage is easier.